code/controllers/subsystem/communications.dm

```diff
@@ -1,67 +1,33 @@
-#define COMMUNICATION_COOLDOWN (30 SECONDS)
-#define COMMUNICATION_COOLDOWN_AI (30 SECONDS)
-#define COMMUNICATION_COOLDOWN_MEETING (5 MINUTES)
+#define COMMUNICATION_COOLDOWN 300
+#define COMMUNICATION_COOLDOWN_AI 300
 
 SUBSYSTEM_DEF(communications)
 	name = "Communications"
 	flags = SS_NO_INIT | SS_NO_FIRE
 
-	COOLDOWN_DECLARE(silicon_message_cooldown)
-	COOLDOWN_DECLARE(nonsilicon_message_cooldown)
-	COOLDOWN_DECLARE(emergency_meeting_cooldown)
+	var/silicon_message_cooldown
+	var/nonsilicon_message_cooldown
 
 /datum/controller/subsystem/communications/proc/can_announce(mob/living/user, is_silicon)
-	if(is_silicon && COOLDOWN_FINISHED(src, silicon_message_cooldown))
-		return TRUE
-	else if(!is_silicon && COOLDOWN_FINISHED(src, nonsilicon_message_cooldown))
-		return TRUE
+	if(is_silicon && silicon_message_cooldown > world.time)
+		. = FALSE
+	else if(!is_silicon && nonsilicon_message_cooldown > world.time)
+		. = FALSE
 	else
-		return FALSE
+		. = TRUE
 
 /datum/controller/subsystem/communications/proc/make_announcement(mob/living/user, is_silicon, input)
 	if(!can_announce(user, is_silicon))
 		return FALSE
 	if(is_silicon)
 		minor_announce(html_decode(input),"[user.name] Announces:")
-		COOLDOWN_START(src, silicon_message_cooldown, COMMUNICATION_COOLDOWN_AI)
+		silicon_message_cooldown = world.time + COMMUNICATION_COOLDOWN_AI
 	else
 		priority_announce(html_decode(user.treat_message(input)), null, 'sound/misc/announce.ogg', "Captain", has_important_message = TRUE)
-		COOLDOWN_START(src, nonsilicon_message_cooldown, COMMUNICATION_COOLDOWN)
+		nonsilicon_message_cooldown = world.time + COMMUNICATION_COOLDOWN
 	user.log_talk(input, LOG_SAY, tag="priority announcement")
 	message_admins("[ADMIN_LOOKUPFLW(user)] has made a priority announcement.")
```

Comment:
other than the emergency meeting stuff these were all code improvements, they shouldnt be reverted

Reply:
stan fikou